src/Symfony/Component/Validator/Violation/ConstraintViolationBuilder.php

```diff
@@ -46,7 +46,7 @@ class ConstraintViolationBuilder implements ConstraintViolationBuilderInterface
     /**
      * @param string $message The error message as a string or a stringable object
      */
-    public function __construct(ConstraintViolationList $violations, Constraint $constraint, $message, array $parameters, $root, $propertyPath, $invalidValue, TranslatorInterface $translator, $translationDomain = null)
+    public function __construct(ConstraintViolationList $violations, ?Constraint $constraint, $message, array $parameters, $root, $propertyPath, $invalidValue, TranslatorInterface $translator, $translationDomain = null)
```

Comment:
constraint must be nullable as it might be null when called in \Symfony\Component\Validator\Context\ExecutionContext::buildViolation. it's also nullable in ConstraintViolation

Reply:
⚠️ When merging this up to 6.0, we need to make the `$constraint` property nullable too.